dr_ensenso_node/src/ensenso.cpp

```diff
@@ -483,6 +487,64 @@ class EnsensoNode: public ros::NodeHandle {
 		return true;
 	}
 
+	bool onGetCameraParams(dr_ensenso_msgs::GetCameraParams::Request & req, dr_ensenso_msgs::GetCameraParams::Response & res) {
+		if (true
+			&& (req.camera) == dr_ensenso_msgs::GetCameraParams::Request::MONO
+			&& (req.camera) == dr_ensenso_msgs::GetCameraParams::Request::LEFT
+			&& (req.camera) == dr_ensenso_msgs::GetCameraParams::Request::RIGHT
+		) {
+			ROS_ERROR_STREAM("Invalid camera type " << req.camera << " requested.");
+			return false;
+		}
+
+		if (req.camera == dr_ensenso_msgs::GetCameraParams::Request::MONO && !ensenso_camera->hasMonocular()) {
+			ROS_ERROR_STREAM("No monocular camera available!");
+			return false;
+		}
+
+		cv::Mat mat;
+		switch (req.type) {
```

Comment:
Could you factor this whole switch statement into it's own function? In this cpp file is fine. An anonymous namespace would be best. Then, instead of doing `mat = ...` each time, you can just return the `mat` directly without having a temporary variable.